### ISSUE TITLE
Are there plans to support Python 3.12.1?

### ISSUE THREAD

**Issue by RicardoEscobar** [capitalone/datacompy#262]
I can't even install it:

```
...
UPDATING build\lib.win-amd64-cpython-312\pandas/_version.py
      set build\lib.win-amd64-cpython-312\pandas/_version.py to '2.0.2'
      running build_ext
      building 'pandas._libs.algos' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pandas
Failed to build pandas
ERROR: Could not build wheels for pandas, which is required to install pyproject.toml-based projects
```

**Comment by fdosani** [comment]
We will support python 3.12. The toml file just needs updating and also there is the question of spark support which currently I'm not sure works with Python 3.12. Just need to work through those questions first, but at a minimum pandas should be able to work. I can take a look at this tomorrow morning and see the LoE involved.

**Comment by fdosani** [comment]
@RicardoEscobar Seems like Spark still doesn't support 3.12 and the same with Ray (Fugue). So we might be stuck without 3.12 support for now.

**Comment by mbrooksuk** [comment]
Hi, just wondering if you have any update on supporting python 3.12.1?

**Comment by fdosani** [comment]
> Hi, just wondering if you have any update on supporting python 3.12.1?

Hey there, We do want to support it. Right now Spark has some compatibility issues. Are you using it for non-Spark use cases?

**Comment by mbrooksuk** [comment]
Hi, yes, purely in a python notebook

**Comment by fdosani** [comment]
Maybe a compromise might be to allow it for Polars and Pandas only until Spark catches up.

**Comment by mbrooksuk** [comment]
Im very new to python, but datacompy is exactly the tool im looking for!  My company only allows python 3.12.1,  how would you allow it for pandas?

**Comment by fdosani** [comment]
> Im very new to python, but datacompy is exactly the tool im looking for! My company only allows python 3.12.1, how would you allow it for pandas?

I'll need to make a change and cut a new release. Will try and come up with something shortly.

**Comment by fdosani** [comment]
Can you provide a output from: `pip freeze` please. Also I just released `0.12.1` if you want to try pip installing that.

**Comment by RicardoEscobar** [comment]
I solved this issue. I'm preparing my answer to this.
Tldr. Installing a previous version of pandas, I dont remember the number.

**Comment by RicardoEscobar** [comment]
> Can you provide a output from: `pip freeze` please. Also I just released `0.12.1` if you want to try pip installing that. Should also note that many times with these types of errors it is some underlying library which is having issues building the package because of missing dependencies like compilers etc.
> 
> I'd also recommend trying to `conda install datacompy`
[YouTube explanation](https://youtu.be/4ng0PBSGB-U?si=gUwBU0NZm92IYFaC)
The relevant entries are `datacompy==0.11.0` and `pandas==2.2.0`, recent versions of pandas create a dependency incompatibility issue.

This is the contents of the `requirements.txt`:
```
$ cat requirements.txt
adagio==0.2.4
antlr4-python3-runtime==4.11.1
appdirs==1.4.4
certifi==2024.2.2
charset-normalizer==3.3.2
comtypes==1.4.2
datacompy==0.11.0
et-xmlfile==1.1.0
fs==2.4.16
fsspec==2024.2.0
fugue==0.8.7
fugue-sql-antlr==0.2.0
greenlet==3.0.3
idna==3.6
Jinja2==3.1.3
MarkupSafe==2.1.5
MouseInfo==0.1.3
numpy==1.26.4
openpyxl==3.1.2
ordered-set==4.1.0
packaging==23.2
pandas==2.2.0
pyarrow==15.0.0
PyAutoGUI==0.9.54
PyGetWindow==0.0.9
PyMsgBox==1.0.9
pymsteams==0.2.2
pyodbc==5.1.0
pyperclip==1.8.2
PyRect==0.2.0
PyScreeze==0.1.30
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
pytweening==1.2.0
pytz==2024.1
pywin32==306
pywinauto==0.6.8
qpd==0.4.4
requests==2.31.0
setuptools==69.1.1
six==1.16.0
SQLAlchemy==2.0.28
sqlglot==22.2.0
tabulate==0.9.0
triad==0.9.5
typing_extensions==4.10.0
tzdata==2024.1
urllib3==2.2.1
wheel==0.42.0
```

**Comment by mbrooksuk** [comment]
> Can you provide a output from: `pip freeze` please. Also I just released `0.12.1` if you want to try pip installing that. Should also note that many times with these types of errors it is some underlying library which is having issues building the package because of missing dependencies like compilers etc.
> 
> I'd also recommend trying to `conda install datacompy`

Hi, this worked for us, since the 0.12.1 release datacompy installed without any dependency errors.  Thanks for sorting this!
